WCFDataService/Client/System/Data/Services/Client/Materialization/ODataEntityMaterializer.cs

```diff
@@ -431,6 +447,8 @@ internal static ODataEntry ProjectionGetEntry(MaterializerEntry entry, string na
             string[] properties,
             Func<object, object, Type, object>[] propertyValues)
         {
+            if (entry == null && materializer.MaterializerContext.AutoNullPropagation)
+                return null;
             if (entry.Entry == null)
```

Comment:
Correct, When AutoNullPropagation is true and server sends a null - "entry" itself will be null, so we only need to check this. The "entry" is returned from Materializer.GetEntry() that returns null if the ODataEntry is null.